service-spec/service-spec-commons/src/main/java/io/hyscale/servicespec/commons/builder/EffectiveServiceSpecBuilder.java

```diff
@@ -36,75 +36,75 @@
         this.fieldMetaDataProvider = fieldMetaDataProvider;
         return this;
     }
 
-	public String build() throws HyscaleException {
-
-		validate(serviceSpec, profile);
-
-		if (this.serviceInputType == ServiceInputType.YAML) {
-			// convert to JSON
-			serviceSpec = ServiceSpecBuilderUtil.yamlToJson(serviceSpec);
-			profile = ServiceSpecBuilderUtil.yamlToJson(profile);
-		}
-
-		// Remove unrequired fields from the profile
-		profile = ServiceSpecBuilderUtil.updateProfile(profile);
-
-		String strategicMergeJson = StrategicPatch.apply(serviceSpec, profile, fieldMetaDataProvider);
-
-		return strategicMergeJson;
-	}
-
-	private void validate(String serviceSpec, String profile) throws HyscaleException {
-		if (StringUtils.isBlank(serviceSpec)) {
-			throw new HyscaleException(ServiceSpecErrorCodes.SERVICE_SPEC_REQUIRED);
-		}
-		if (StringUtils.isBlank(profile)) {
-			throw new HyscaleException(ServiceSpecErrorCodes.SERVICE_PROFILE_REQUIRED);
-		}
-		ObjectMapper mapper;
-		if (ServiceInputType.JSON == this.serviceInputType) {
-			mapper = ObjectMapperFactory.jsonMapper();
-		} else {
-			mapper = ObjectMapperFactory.yamlMapper();
-		}
-		try {
-			mapper.readTree(serviceSpec);
-		} catch (IOException e) {
-			throw new HyscaleException(ServiceSpecErrorCodes.SERVICE_SPEC_PARSE_ERROR);
-		}
-		try {
-			mapper.readTree(profile);
-		} catch (IOException e) {
-			throw new HyscaleException(ServiceSpecErrorCodes.SERVICE_PROFILE_PARSE_ERROR);
-		}
-	}
+    public String build() throws HyscaleException {
+
+        validate(serviceSpec, profile);
+
+        if (this.serviceInputType == ServiceInputType.YAML) {
+            // convert to JSON
+            serviceSpec = YamlToJsonConverter.yamlToJson(serviceSpec);
+            profile = YamlToJsonConverter.yamlToJson(profile);
+        }
+
+        // Remove unrequired fields from the profile
+        profile = ServiceSpecBuilderUtil.updateProfile(profile);
+
+        String strategicMergeJson = StrategicPatch.apply(serviceSpec, profile, fieldMetaDataProvider);
+
+        return strategicMergeJson;
+    }
+
+    private void validate(String serviceSpec, String profile) throws HyscaleException {
+        if (StringUtils.isBlank(serviceSpec)) {
+            throw new HyscaleException(ServiceSpecErrorCodes.SERVICE_SPEC_REQUIRED);
+        }
+        if (StringUtils.isBlank(profile)) {
+            throw new HyscaleException(ServiceSpecErrorCodes.SERVICE_PROFILE_REQUIRED);
+        }
+        ObjectMapper mapper;
+        if (ServiceInputType.JSON == this.serviceInputType) {
+            mapper = ObjectMapperFactory.jsonMapper();
+        } else {
+            mapper = ObjectMapperFactory.yamlMapper();
+        }
+        try {
+            mapper.readTree(serviceSpec);
+        } catch (IOException e) {
+            throw new HyscaleException(ServiceSpecErrorCodes.SERVICE_SPEC_PARSE_ERROR);
```

Comment:
No formatiing changes should be checked in . Kindly cross check before commiting